uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,447 @@ define([
     templateUrl: ['settings', function (settings) {
       return settings.pathTpl + 'components/manager-leave-requests.html';
     }],
-    controllerAs: 'requests',
-    controller: ['$log', controller]
+    controllerAs: 'ctrl',
+    controller: ['$log', '$q', 'Contact', 'AbsencePeriod', 'AbsenceType', 'LeaveRequest',
+      'OptionGroup', controller]
   });
 
 
-  function controller($log) {
+  function controller($log, $q, Contact, AbsencePeriod, AbsenceType, LeaveRequest, OptionGroup) {
     $log.debug('Component: manager-leave-requests');
+
+    var vm = Object.create(this),
+      allStatus = {
+        name: 'all',
+        label: 'All'
+      };
+
+    vm.isFilterExpanded = false;
+    vm.absencePeriods = [];
+    vm.filteredUsers = [];
+    vm.absenceTypes = [];
+    //leaveRequests.table - to handle table data
+    //leaveRequests.filter - to handle left nav filter data
+    vm.leaveRequests = {
+      table: {
+        list: []
+      },
+      filter: {
+        list: []
+      }
+    };
+    vm.filters = {
+      contactFilters: {
+        region: null,
+        department: null,
+        level_type: [],
+        location: null
+      },
+      leaveRequestFilters: {
+        selectedUsers: null,
+        selectedPeriod: null,
+        selectedAbsenceTypes: null,
+        leaveStatus: allStatus,
+        pending_requests: false
+      }
+    };
+    vm.pagination = {
+      page: 1,
+      size: 7
+    };
+    vm.loading = {
+      content: true,
+      page: true
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    vm.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Returns the label of a Leave request when id is given
+     *
+     * @param {string} id - id of the leave request
+     * @return {string}
+     */
+    vm.getLeaveStatusByValue = function (id) {
+      if (vm.leaveRequestStatuses.length) {
+        return vm.leaveRequestStatuses.find(function (status) {
+          return status.value == id;
+        }).label;
+      }
+    };
+
+    /**
+     * Returns the title of a Absence type when id is given
+     *
+     * @param {string} id - id of the Absence type
+     * @return {string}
+     */
+    vm.getAbsenceTypesByID = function (id) {
```

Comment:
Do you really need this thought? I see that you are using it in the view
```html
<td style="width: 20%;">{{ctrl.getAbsenceTypesByID(request.type_id)}}</td>
```
but what if you simply have vm.absenceTypes accessible by ids (see [here](https://github.com/civicrm/civihr/blob/la-milestone-3/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-report.js#L257-L262)? You could simply do then
```html
<td style="width: 20%;">{{ctrl.absenceTypes[request.type_id].title}}</td>
```
the code would be simpler, resulting in less moving parts and stuff to test